packages/react/src/components/Toolbar/ToolbarItem.tsx

```diff
@@ -101,6 +120,20 @@ class ToolbarItem extends UIComponent<WithAsProp<ToolbarItemProps>, ToolbarItemS
         {childrenExist(children) ? children : Icon.create(icon)}
       </ElementType>
     )
+
+    if (popup) {
+      return Popup.create(popup, {
+        defaultProps: {
+          accessibility: popupFocusTrapBehavior,
+        },
+        overrideProps: {
+          trigger: renderedItem,
+          children: undefined, // force-reset `children` defined for `Popup` as it collides with the `trigger
```

Comment:
```suggestion
          children: undefined, // force-reset `children` defined for `Popup` as it collides with the `trigger`
```

Reply:
Still not sure why we need this explicitly, let me fetch the branch and play with it a bit.

Reply:
I don't see any issue without this being applied, as the user cannot specify children, and we are not passing it. Probably I am missing something..

Reply:
- Without this, user can define the item as follows:
```jsx
{
  key: 'font-color',
  icon: { name: 'font-color', outline: true },
  popup: {
    content: { content: <Input icon="search" placeholder="Search..." /> },
    children: <Button content="invalid" />,
  },
},
```
- `ToolbarItem` uses `Popup.trigger` to 'point to itself'.
- But `children` has priority over `trigger` in `Popup` implementation.
- That means that **invalid** button would be rendered instead of the **font-color** icon.

That's the reason for the force-reset.

Reply:
Got it, I thought the typings would be enough, but I see your point.